### PR TITLE
Gonzalo/pokerhands

### PR DESCRIPTION
**Pull request by dulphyj**

### CODE REVIEW

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Compare.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        boolean verified = false;
+        majorCard = valueCard.converterCharSuit(hand);
+        if (majorCard[card1] == majorCard[card2] && majorCard[card1] == majorCard[card3]
+            && majorCard[card1] == majorCard[card4] && majorCard[card1] == majorCard[card5]) {
+            verified = true;
+        }
+        return verified;
+    }
+
+    public boolean compareCard1(String[] hand) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Compare.java:26]: This is good. Try to improve it by avoiding repeating code in `compareCard2`, `compareCard3` and so on. You should be able to have one single method `compareCard` that you could reuse for all cards

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Compare.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public boolean compareCard4(String[] hand) {
+        boolean verified = false;
+        majorCard = valueCard.converterCharValue(hand);
+        if (majorCard[card4] != majorCard[card5]) {
+            verified = true;
+        }
+        return verified;
+    }
+
+    public boolean isStraight(String[] hand) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Compare.java:64]: cool that you are using boolean!

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Cards.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.fundacionjala.at15.katas.pokerhands.gonzalo;
+
+import java.util.List;
+
+public abstract class Cards {
+
+    public abstract List<String> cards(String suit);
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Cards.java:7]: Always try to use a verb for method names, this way it helps to describe what the method is doing (in this case, does it get cards? does it read cards? does it create cards?)

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Compare.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        boolean verified = false;
+        majorCard = valueCard.converterCharValue(hand);
+        if (majorCard[card2] != majorCard[card3] && majorCard[card2] != majorCard[card4]
+            && majorCard[card2] != majorCard[card5]) {
+            verified = true;
+        }
+        return verified;
+    }
+
+    public boolean compareCard3(String[] hand) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Compare.java:46]: by the return type, looks like the name of this method should be improved. Yes, it compares two cards, but what is comparing? and what does the result mean? for what I read in the code, I assume this chould be `doesCardsHaveTheSameValue(...)` or `isHandAStraight(...)`

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Converter.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
+package org.fundacionjala.at15.katas.pokerhands.gonzalo;
+
+public class Converter {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Converter.java:3]: Good name

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Converter.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    private final int cardSuit = 1;
+    private char valueThisCard;
+    private char[] cardsValue;
+    private int[] cardsValueInt;
+    private int auxMax = 0;
+    private int auxMin = 0;
+    private int major = 0;
+    private int less;
+
+    public char[] converterCharValue(String[] hand) {
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/Converter.java:15]: Good name, I would improve it to `convertValuesToChar()`. Same for suit and int

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/DeckOfCards.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package org.fundacionjala.at15.katas.pokerhands.gonzalo;
+
+import java.util.List;
+
+public class DeckOfCards {
+    private final String clubs = "C";
+    private final String diamonds = "D";
+    private final String hearts = "H";
+    private final String spades = "S";
+    private Cards suit;
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/DeckOfCards.java:10]: Nice

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/ValueHandFlush.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package org.fundacionjala.at15.katas.pokerhands.gonzalo;
+
+public class ValueHandFlush extends RankedHand {
+    private Compare compare = new Compare();
+    private String isFlush;
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/ValueHandFlush.java:5]: Looks like this should be a boolean

File: src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/ValueHandStraightFlush.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package org.fundacionjala.at15.katas.pokerhands.gonzalo;
+
+public class ValueHandStraightFlush extends RankedHand {
+    private Compare compare = new Compare();
+    private String isFlush;
```
Comment (J0c4) [inline src/main/java/org/fundacionjala/at15/katas/pokerhands/gonzalo/ValueHandStraightFlush.java:5]: boolean?

File: src/test/java/org/fundacionjala/at15/katas/pokerHands/gonzalo/CompareTest.java [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    }
+    @Test
+    public void itShouldBeNoFlush() {
+        String[] hand = {"2H", "6H", "AS", "KH", "5H"};
+        boolean expected = false;
+        Compare compare = new Compare();
+        assertEquals(expected, compare.compareSuit(hand));
+    }
+
+
```
Comment (J0c4) [inline src/test/java/org/fundacionjala/at15/katas/pokerHands/gonzalo/CompareTest.java:24]: Please remove empty lines that do not help to improve code readability

File: src/test/java/org/fundacionjala/at15/katas/pokerHands/gonzalo/CompareTest.java [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+        boolean expected = true;
+        Compare compare = new Compare();
+        assertEquals(expected, compare.isStraight(hand));
+    }
+    @Test
+    public void itShouldBeNoStraight() {
+        String[] hand = {"2C", "6D", "AH", "3H", "5C"};
+        boolean expected = false;
+        Compare compare = new Compare();
+        assertEquals(expected, compare.isStraight(hand));
```
Comment (J0c4) [inline src/test/java/org/fundacionjala/at15/katas/pokerHands/gonzalo/CompareTest.java:38]: you could use `assertTrue` instead in order to simplify

File: src/test/java/org/fundacionjala/at15/katas/pokerHands/gonzalo/CompareTest.java [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    }
+    @Test
+    public void itShouldBeEqualsCardsCompareCard2() {
+        String[] hand = {"2C", "6D", "AH", "6H", "5C"};
+        boolean expected = false;
+        Compare compare = new Compare();
+        assertEquals(expected, compare.compareCard2(hand));
+    }
+
+
```
Comment (J0c4) [inline src/test/java/org/fundacionjala/at15/katas/pokerHands/gonzalo/CompareTest.java:75]: unnecessary empty lines

File: src/test/java/org/fundacionjala/at15/katas/pokerHands/gonzalo/ValueHandStraightTest.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
+package org.fundacionjala.at15.katas.pokerhands.gonzalo;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class ValueHandStraightTest {
```
Comment (J0c4) [inline src/test/java/org/fundacionjala/at15/katas/pokerHands/gonzalo/ValueHandStraightTest.java:7]: Nice tests

**Review by J0c4 — CHANGES_REQUESTED** [review]

**Comment by dulphyj** [comment]
I'm not sure about the last two classes added, DealCards and Main, I don't know how to test when I use random numbers, in theory it already works, I just can't check between hands with the same game which one is worth more Eg "AS AH AC 2S 3H " And "7S 7H 7C KS AC" should win the first hand, but my result is "equal"

**Comment by J0c4** [comment]
The way the logic is separated it helps to have smaller pieces that are easier to review and understand and in general you did a good job identifying possible classes with specific responsability

What it needs to be improved is how you define your classes (specially the names) so when you create an object from the classes, they make sense when you read their state and behavior. For example, you have 
```
CardSuit cardSuit = new CardSuit()
cardSuit.cards(...)
```
Which would be read as "an object cardSuit do cards". Instead, if method name is improved, you could have something like
```
CardSuit cardSuit = new CardSuit()
cardSuit.createCards(...)
```
Then, it can be read as "an object cardSuit creates cards", which reflects the name and what is that object doing when you call that specific method.

**Review by J0c4 — APPROVED** [review]

**Pull request merged**